node/silkworm/db/storage.hpp

```diff
@@ -15,88 +15,92 @@
+using PruneDistance = std::optional<BlockNum>;   // for 'older' type
+using PruneThreshold = std::optional<BlockNum>;  // for 'before' type
 
 inline constexpr BlockNum kDefaultPruneThreshold{90'000};
 
-struct PruneMode {
-    bool initialized{false};    // Whether db storage has been initialized
-    PruneDistance history;      // Amount of blocks for history to keep in db
-    PruneDistance receipts;     // Amount of blocks for receipts to keep in db
-    PruneDistance tx_index;     // Amount of blocks for tx_index to keep in db
-    PruneDistance call_traces;  // Amount of blocks for call_traces to keep in db
-
-    [[nodiscard]] std::string to_string() const {
-        if (!initialized) {
-            return "default";
-        }
-
-        std::string short_form{"--prune="};
-        std::string long_form{};
-
-        if (history.has_value()) {
-            if (history.value() == kDefaultPruneThreshold) {
-                short_form += "h";
-            } else {
-                long_form += (" --prune.h.older=" + std::to_string(history.value()));
-            }
-        }
-
-        if (receipts.has_value()) {
-            if (receipts.value() == kDefaultPruneThreshold) {
-                short_form += "r";
-            } else {
-                long_form += (" --prune.r.older=" + std::to_string(receipts.value()));
-            }
-        }
-
-        if (tx_index.has_value()) {
-            if (tx_index.value() == kDefaultPruneThreshold) {
-                short_form += "t";
-            } else {
-                long_form += (" --prune.t.older=" + std::to_string(tx_index.value()));
-            }
-        }
-
-        if (call_traces.has_value()) {
-            if (call_traces.value() == kDefaultPruneThreshold) {
-                short_form += "c";
-            } else {
-                long_form += (" --prune.c.older=" + std::to_string(call_traces.value()));
-            }
-        }
-
-        return short_form + long_form;
+class BlockAmount {
+  public:
+    enum Type {
+        kOlder,  // Prune Data Older than (moving window)
+        kBefore  // Prune data before (fixed)
+    };
+    explicit BlockAmount() : value_{std::nullopt}, enabled_{false}, type_{Type::kOlder} {};
+    explicit BlockAmount(Type type, BlockNum value) : value_{value}, enabled_{true}, type_{type} {};
+
+    [[nodiscard]] bool enabled() const { return enabled_; }
+    [[nodiscard]] Type type() const { return type_; };
+    [[nodiscard]] BlockNum value() const;
+    [[nodiscard]] BlockNum value_from_head(BlockNum stage_head) const;
+
+    void to_string(std::string& short_form, std::string& long_form, char prefix) const;
+
+    bool operator==(const BlockAmount& other) const {
+        return enabled_ == other.enabled_ && type_ == other.type_ && value_ == other.value_;
     }
+    bool operator!=(const BlockAmount& other) const { return !(this->operator==(other)); }
+
+  private:
+    const std::optional<BlockNum> value_;
```

Comment:
`const` data member are a bit of a headache in C++: https://lesleylai.info/en/const-and-reference-member-variables. TBH, I wouldn't bother and just use the simplest approach
```
struct Point {
    int x = 0;
    int y = 0;
};

const Point immutable_point {42, 55};
```